### DISCUSSION TITLE
Is there any way that I can add routes to "/api" prefix?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#990] [Q&A]
[I see that the api group is created here](https://github.com/pocketbase/pocketbase/blob/30d1b9f358af168b3612f84a72fc48571220167a/apis/base.go#L86)

When extending Pocketbase, is there any way that I can add additional routes to the "/api" prefix, or is it closed?  I looked through Echo and Pocketbase source but I'm not sure since this might catch everything.

```javascript
	// catch all any route
	api.Any("/*", func(c echo.Context) error {
		return echo.ErrNotFound
	}, ActivityLogger(app))
```

**Comment by ganigeorgiev** [comment]
> When extending Pocketbase, is there any way that I can add additional routes to the "/api" prefix, or is it closed?

You can register custom routes to the `/api` prefix as long as they don't duplicate with the existing one. This is why in the upcoming v0.8.0 documentation there is a recommendation to use a custom prefix to avoid collisions with future internal routes. Here is an example that registers a `/api/hello` route:

```go
package main

import (
	"log"
	"net/http"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	        // you can also use the shorter e.Router.GET(...)
		e.Router.AddRoute(echo.Route{
			Method: http.MethodGet,
			Path:   "/api/hello",
			Handler: func(c echo.Context) error {
				return c.String(200, "Hello world!")
			},
			Middlewares: []echo.MiddlewareFunc{
				apis.RequireAdminAuth(),
			},
		})

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

**Reply by dschissler** [reply]
@ganigeorgiev Have you settled on a style for the custom prefix yet?  I'm thinking either prefix `/api/ccc` where the c's are abbreviation of project name, or `/apiv1`.  This is probably a pedantic question, but what are people using for their prefix?
